great_expectations/data_context/types/base.py

```diff
@@ -1058,14 +1020,31 @@ def from_commented_map(cls, commented_map):
             )
             raise
 
-    def to_yaml(self, outfile):
-        commented_map = deepcopy(self.commented_map)
+    def get_schema_validated_updated_commented_map(self) -> CommentedMap:
+        commented_map: CommentedMap = deepcopy(self.commented_map)
         commented_map.update(dataContextConfigSchema.dump(self))
-        yaml.dump(commented_map, outfile)
+        return commented_map
+
+    def to_yaml(self, outfile):
+        """
+        :returns None (but writes a YAML file containing the project configuration)
+        """
+        yaml.dump(self.get_schema_validated_updated_commented_map(), outfile)
+
+    def to_yaml_str(self) -> str:
+        """
+        :returns a YAML string containing the project configuration
+        """
+        return object_to_yaml_str(self.get_schema_validated_updated_commented_map())
+
+    def to_dict(self) -> dict:
```

Comment:
```suggestion
    def to_json_dict(self) -> dict:
```

Reply:
Related: instead of inheriting from DictDot, could we inherit from SerializableDictDot now?

Reply:
This was done for DataContextConfig (committed).